sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py

```diff
@@ -122,6 +124,35 @@ def set_message_partition_key(message, partition_key):
         message.header = header
 
 
+@contextmanager
+def send_context_manager():
+    span_impl_type = (
+        settings.tracing_implementation()
+    )  # type: Type[AbstractSpan]
+
+    if span_impl_type is not None:
+        with span_impl_type(name="Azure.EventHubs.send") as child:
+            child.kind = SpanKind.CLIENT
+            yield child
+    else:
+        yield None
+
+
+def add_link_to_send(event, send_span):
+    """Add Diagnostic-Id from event to span as link.
+    """
+    try:
+        if send_span:
+            if event.properties:
+                traceparent = event.properties.get(b"Diagnostic-Id", "").decode("ascii")
+            else:
+                traceparent = ""
+
+            send_span.link(traceparent)
```

Comment:
What are the semantics of linking to the traceparent `""`? And what happens if you do it multiple times?

Reply:
Fair, I should link only inside the if

Reply:
Fixed